### ISSUE TITLE
Cannot import in a TypeScript project

### ISSUE THREAD

**Issue by DOEHOONLEE** [Eric-Canas/Homography.js#2]
Hi.
I tested importing Homography class in a Next.js based project, but just required an extra step like this

```js
declare module 'homography'
```

and it worked fine.

However, with the same configuration, setup, and the above declaration in a TypeScript project, importing Homography fails.

Is there some way for me to use it in a TS project?

Thanks!

**Comment by Eric-Canas** [comment]
Hi @DOEHOONLEE 

I have uploaded a new version to npm, could you try to just ```npm install --save homography``` again and run with the new version?

**Comment by Eric-Canas** [comment]
@DOEHOONLEE Yes, sorry,

Could you try it again with the new version I have just uploaded? It should be solved now (I hope).

At the very beginning, I thought that it would a good idea to upload the npm version as a Node.js module, for using it in the backend. Later, I realized that it was a never ever use case.

**Comment by DOEHOONLEE** [comment]
I'm sorry for the late reply. I did use the latest version you updated to (1.5.0).
I actually am using it on the back-end side. Should it be a problem? It does work on Next.js server-side though.

**Comment by Eric-Canas** [comment]
I thought version 1.5.0 was the one giving you problems the last time, the new one I uploaded was 1.8.0.

Anyway, I'm not very experienced with TypeScript projects, so I don't really know how could you include those ```@types/module```. But I don't think it should be difficult. The documentation within the source code does already include all the expected types for each parameter, so I suppose that you would just need to parse it. 

About getting it as base64, I think you could just use the method ```HTMLImageElementFromImageData``` of the library. It takes the ```ImageData``` as input and returns an ```<img/>``` element, whose ```src``` is actually the image as a _base64 image/png_.

**Comment by DOEHOONLEE** [comment]
> I thought version 1.5.0 was the one giving you problems the last time, the new one I uploaded was 1.8.0.

oh, I see I see. Let me try again with 1.8.0 :)

> About getting it as base64, I think you could just use the method HTMLImageElementFromImageData of the library. It takes the ImageData as input and returns an <img/> element, whose src is actually the image as a base64 image/png.

I see that the result comes with buffer data and I was trying to convert it into base64 somehow and it's all done on the server-side so I can't use the `<img/>` tag

**Comment by Eric-Canas** [comment]
There is a [canvas library in node](https://www.npmjs.com/package/canvas). I think you could use it for that purpose. It should work quite similar to frontend `<canvas>`. You could just write the `ImageData` in the `canvas` and then export it to [data URL](https://www.npmjs.com/package/canvas#canvastodataurl)

**Comment by DOEHOONLEE** [comment]
oh, I see!

Yes. I've been using node-canvas, but had a trouble converting buffer to base64 or drawing on canvas.

The output is like below

```
{
  pngImage: Stream {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    width: 998,
    height: 998,
    data: <Buffer 00 48 ff ff 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ... 3983966 more bytes>,
    gamma: 0,
    writable: true,
    readable: true,
    _parser: Stream {
      _events: [Object: null prototype],
      _eventsCount: 5,
      _maxListeners: undefined,
      _buffers: [],
      _buffered: 0,
      _reads: [Array],
      _paused: false,
      _encoding: 'utf8',
      writable: true,
      _parser: [Object],
      _options: [Object],
      [Symbol(kCapture)]: false
    },
    _packer: Stream {
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      _packer: [Object],
      _deflate: [Deflate],
      readable: true,
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false
  },
  output_img: Uint8ClampedArray(3984016) [
    0, 72, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0,
```

and I tried to convert this to base64 like

```
const base64 = Buffer(result.pngImage.data).toString('base64')
```

then, I get infinite lines of codes on the terminal :(

**Comment by Eric-Canas** [comment]
:(

You got that problem by using the `toDataURL('image/png')` method?  That's how I get it on the frontend, and I assume it should work, more or less, the same way on the backend.

**Comment by DOEHOONLEE** [comment]
I found this for those who want to convert the PNG stream to base64

```
const streamToBase64 = (stream:any) => {
  const concat = require('concat-stream')
  const { Base64Encode } = require('base64-stream')

  return new Promise((resolve, reject) => {
    const base64 = new Base64Encode()

    const cbConcat = (base64:any) => {
      resolve(base64)
    }

    stream
      .pipe(base64)
      .pipe(concat(cbConcat))
      .on('error', (error:any) => {
        reject(error)
      })
  })
}
```
